--- tinyos-2.1.2/tools/tinyos/misc/tos-locate-jre.~1~	2012-08-21 03:28:42.000000000 +0900
+++ tinyos-2.1.2/tools/tinyos/misc/tos-locate-jre	2016-12-03 01:41:44.049111352 +0900
@@ -67,11 +67,16 @@
 
     pn=`sw_vers -productName | sed 's/ //g'`
     pv=`sw_vers -productVersion | awk -F. '{ print $1 "." $2 }'`
+    if [[ $pv = 10.6 ]]; then # Snow Leopard
     ## Since we only want to modify this one script for now, stick the
     ## extra subdir at the end, beceuase that's what the reset of scripts
     ## presume. This is a work-around and should be eliminated eventually.
     xcode_jdk=/Developer/SDKs/${pn}${pv}.sdk/System/Library/Frameworks/JavaVM.framework/Versions/A/Headers
     jhome=/Library/Java/Home
+    else
+    jhome=/System/Library/Frameworks/JavaVM.framework/Versions/Current
+    xcode_jdk=$jhome/Commands
+    fi
     ;;
 
     Linux)
@@ -134,9 +139,15 @@
     # Windows, and IBM Java: in jre/bin
     # Sun Java on Linux: in jre/lib/i386
     if [ `uname` = "Darwin" ]; then 
+	if [[ $pv = 10.6 ]]; then # Snow Leopard
 	jnilocate "/Library/Java/Extensions"
+	else
+	jnilocate $jhome
+	fi
     elif "$jhome/bin/java" -version 2>&1 | grep -q IBM || cygpath -w / >/dev/null 2>/dev/null; then
 	jnilocate "$jhome/jre/bin" || jnilocate "$jhome/bin"
+    elif "$jhome/bin/java" -version 2>&1 | grep -qi openjdk; then
+	jnilocate "$jhome/include"
     else
 	arch=`uname -m`
 	jnilocate "$jhome/jre/lib/$arch" || \
